requirements/requirements.txt

```diff
@@ -2,3 +2,4 @@ git+https://github.com/facebookresearch/hydra#subdirectory=tools/configen
 git+https://github.com/facebookresearch/hydra
 torch==1.6.0
 torchvision==0.7.0
+scipy
```

Comment:
We shouldn't make the primary requirements.txt the union of the dependencies of all subprojects.

Reply:
yeah we need to migrate this out of here. I think we should generally have per project dependencies for both using the library and running tests / doing dev.

It's not clear that someone who wants to use torchvision configs will ever use a model that depends on scipy for example, so we shouldn't force them to install it. But if they plan to do dev/run tests, then it makes sense.

Reply:
Otherwise I think this PR looks good.

Let's merge and then I can follow up with a reorganization of how we manage dependencies across projects in a new PR.

Reply:
The tests seems a bit too shallow to me. Is it really the best we can do to verify that the instantiated objects are good?

Reply:
I agree that the tests are minimal. 

@shivamdb, would it be possible to check that both:
1. the models instantiate correctly (which you've already done) 
2. each instantiated model is capable of running a single `forward()` pass?

Reply:
About the dependencies:
another approach is to include separate put things needed to run the tests in a requiremenets/dev.txt.

Reply:
Yes, that's the plan. We can have `requirements/dev.txt` for each project. I can update nox to create a specific environment when testing a specific project.

Reply:
As long as we have a handful of projects, we can also have a unified dev dependencies for all of them.

Reply:
Addressing the above comments in #59